libs/jsonforms-components/src/lib/Controls/AddressLookup/AddressLookUpControl.tsx

```diff
@@ -112,40 +119,120 @@ export const AddressLookUpControl = (props: AddressLookUpProps): JSX.Element =>
     setSuggestions([]);
     setErrors({});
   };
+
   const handleRequiredFieldBlur = (name: string) => {
-    if ((!data?.[name] || data?.[name] === '') && requiredFields?.includes(name)) {
-      const err = { ...errors };
-      err[name] = name === 'municipality' ? `city is required` : `${name} is required`;
+    const err = { ...errors };
+    if(data?.["city"] === undefined || data?.["city"] === ""){
+      err[name] = name === 'municipality' ? 'city is required' : ""
       setErrors(err);
-    } else {
+    }
+
+    if(!data?.[name] || data[name] === '' || data?.[name] === undefined){
+      err[name] = name === 'municipality' ? 'city is required' : `${name} is required`;
+      setErrors(err);
+    }
+
+    if(!data?.[name]){
+      err[name] = name === 'addressLine1' ? `${name} is required` : ``;
+      setErrors(err);
+    }
+
+    else{
       delete errors[name];
     }
+
+    setTimeout(() => {
+      setSuggestions([])
+      setOpen(false)
+    }, 100);
   };
+
+  useEffect(() => {
+    if (dropdownRef.current) {
+        const selectedItem = dropdownRef.current.children[selectedIndex];
+        if (selectedItem) {
+          selectedItem.scrollIntoView({
+            block: 'nearest',
+            behavior: 'smooth',
+          });
+        }
+      }
+    }, [selectedIndex, open]);
+
+const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
```

Comment:
@Sankeerth-Mulkalla-goa
Can you add this line
 /* istanbul ignore next */

on top of the function handleKeyDown to ignore this block of code for the code coverage